saleor/order/signals.py

```diff
@@ -14,7 +15,9 @@ def order_status_change(sender, instance, **kwargs):
         order.create_history_entry(
             content=pgettext_lazy(
                 'Order status history entry', 'Order fully paid'))
-        instance.send_confirmation_email()
+        email_data = collect_data_for_email(order)
+        email_data.update({'order_pk': order.pk})
```

Comment:
If `order` is already passed to `collect_data_for_email`, is `email_data.update()` outside of it necessary?

Reply:
Good point. I was planning initially using `collect_data_for_email` for both `send_order_confirmation` and `send_payment_confirmation` and maybe even `send_note_confirmation` (I think I can do it in this PR as a additional improvement). As last two use only `email` and `url`, the first one would be a special case where we need additionally `order_pk`, so updating would be required.

Reply:
Do we actually need `collect_data_for_email`? This function seems to be used always together with functions that send emails, so now sending an email requires calling two functions instead of one. Also, I think all of the order emails functions in `saleor/order/emails` could simply take `order` as a single parameter and construct their contexts inside them based on the order.

Reply:
@elwoodxblues I'vie tried this approach with one function taking `order` as a parameter, but got an error about `Order` object not being JSON serializable. It seems to be Celery related, see this SO answer:
https://stackoverflow.com/a/20826946/3917230